express-notes-api/index.js

```diff
@@ -0,0 +1,102 @@
+const data = require('./data.json');
+const express = require('express');
+const app = express();
+const port = 3000;
+const write = require('./write');
+const fs = require('fs');
+const path = './data.json';
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  res.json(data);
+}
+);
+
+app.get('/api/notes/:id', (req, res) => {
+  const noteAddress = req.params.id;
+  if (noteAddress < 1) {
+    res.status(400).json('error: must be a positive integer');
+  } else if (!data.notes[noteAddress]) {
+    res.status(404).json(`error: cannot find note with id: ${noteAddress}`);
+  }
+  res.status(200).json(data.notes[noteAddress]);
```

Comment:
This line will always execute, which is not what you want. Use `else` or `return` in your if blocks.